core/src/main/java/org/mule/runtime/core/routing/UntilSuccessful.java

```diff
@@ -91,12 +91,13 @@ public void initialise() throws InitialisationException {
     }
     this.untilSuccessfulStrategy.setUntilSuccessfulConfiguration(this);
 
-    if (untilSuccessfulStrategy instanceof Initialisable) {
-      ((Initialisable) untilSuccessfulStrategy).initialise();
-    }
     if (untilSuccessfulStrategy instanceof MuleContextAware) {
       ((MuleContextAware) untilSuccessfulStrategy).setMuleContext(muleContext);
     }
+
+    if (untilSuccessfulStrategy instanceof Initialisable) {
+      ((Initialisable) untilSuccessfulStrategy).initialise();
```

Comment:
why not call `LifecycleUtils#initializeIfNeeded()`? It has a method that receives the muleContext to set.